### ISSUE TITLE
Podman 5.3.0 win-sshproxy.tid: The system cannot find the file specified. (install via scoop)

### ISSUE THREAD

**Issue by johnnykang** [containers/podman#24557]
### Issue Description

after upgrading to the Podman 5.3.0 (via scoop), when starting the podman machine, there is a message 

API forwarding for Docker API clients is not available due to the following startup failures.
        The system cannot find the path specified.

it turned out that the c:\<user_profile>\.local\share\containers\podman\machine\wsl\podman-machine-default\win-sshproxy.tid  is not created.

Downgrading to Podman 5.2.5 , the message is gone.

Podman machine runs in WSL2.

### Steps to reproduce the issue

As above in the description

### Describe the results you received

As above in the description

### Describe the results you expected

podman machine starts and API forwarding for Docker API  clients on Windows machine works as expected without error message.

### podman info output

```yaml
Podman 5.3.0
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by glektarssza** [comment]
Calling `podman machine stop` will also produce the following warning/error in this situation:

```
Could not stop API forwarding service (win-sshproxy.exe): open C:\Users\<username>\.local\share\containers\podman\machine\wsl\podman-machine-default\win-sshproxy.tid: The system cannot find the file specified.
```

**Comment by baude** [comment]
It helps when you provide `podman info` like the template recommends.  In both cases, is it safe to assume WSL is being used?

**Comment by baude** [comment]
@l0rd thoughts ?

**Comment by glektarssza** [comment]
In my case, yes. My `podman machine` instance is running inside of WSL.

**Comment by glektarssza** [comment]
`podman version` output:

```
Client:       Podman Engine
Version:      5.2.5
API Version:  5.2.5
Go Version:   go1.23.2
Git Commit:   10c5aa720d59480bc7edad347c1f5d5b75d4424f
Built:        Wed Oct 23 11:54:08 2024
OS/Arch:      windows/amd64

Server:       Podman Engine
Version:      5.2.5
API Version:  5.2.5
Go Version:   go1.22.7
Built:        Thu Oct 24 18:00:00 2024
OS/Arch:      linux/amd64
```

`podman info` output:

```
host:
  arch: amd64
  buildahVersion: 1.37.5
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.12-2.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: '
  cpuUtilization:
    idlePercent: 99.34
    systemPercent: 0.43
    userPercent: 0.24
  cpus: 4
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "40"
  eventLogger: journald
  freeLocks: 2048
  hostname: GlekPC
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
  kernel: 5.15.167.4-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 16080793600
  memTotal: 16776019968
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.12.2-2.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.12.2
    package: netavark-1.12.2-1.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.12.2
  ociRuntime:
    name: crun
    package: crun-1.17-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.17
      commit: 000fa0d4eeed8938301f3bcf8206405315bc1017
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240906.g6b38f07-1.fc40.x86_64
    version: |
      pasta 0^20240906.g6b38f07-1.fc40.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 4294967296
  swapTotal: 4294967296
  uptime: 0h 2m 15.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 914518016
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 5.2.5
  Built: 1729814400
  BuiltTime: Thu Oct 24 18:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.7
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.5
```

`podman machine info` output:

```
host:
    arch: amd64
    currentmachine: podman-machine-default
    defaultmachine: podman-machine-default
    eventsdir: C:\Users\roblo\.local\share\containers\podman\podman
    machineconfigdir: C:\Users\roblo\.config\containers\podman\machine\wsl
    machineimagedir: C:\Users\roblo\.local\share\containers\podman\machine\wsl
    machinestate: Running
    numberofmachines: 1
    os: windows
    vmtype: wsl
version:
    apiversion: 5.2.5
    version: 5.2.5
    goversion: go1.23.2
    gitcommit: 10c5aa720d59480bc7edad347c1f5d5b75d4424f
    builttime: Wed Oct 23 11:54:08 2024
    built: 1729706048
    osarch: windows/amd64
    os: windows
```

`podman machine inspect` output:

```
[
     {
          "ConfigDir": {
               "Path": "C:\\Users\\roblo\\.config\\containers\\podman\\machine\\wsl"
          },
          "ConnectionInfo": {
               "PodmanSocket": null,
               "PodmanPipe": {
                    "Path": "\\\\.\\pipe\\podman-machine-default"
               }
          },
          "Created": "2024-11-13T14:54:20.3775309-07:00",
          "LastUp": "2024-11-13T14:55:18.3645367-07:00",
          "Name": "podman-machine-default",
          "Resources": {
               "CPUs": 16,
               "DiskSize": 100,
               "Memory": 2048,
               "USBs": []
          },
          "SSHConfig": {
               "IdentityPath": "C:\\Users\\roblo\\.local\\share\\containers\\podman\\machine\\machine",
               "Port": 50275,
               "RemoteUsername": "user"
          },
          "State": "running",
          "UserModeNetworking": false,
          "Rootful": false,
          "Rosetta": false
     }
]
```

**Comment by johnnykang** [comment]
> It helps when you provide `podman info` like the template recommends. In both cases, is it safe to assume WSL is being used?

Yup. Podman machine runs in the WSL2.

**Comment by johnnykang** [comment]
> Also might be worth mentioning that my podman setup was installed via `scoop`. Shouldn't impact anything (I think) but you never know.

Great point. And that's my installation method (via scoop) as well.

**Comment by l0rd** [comment]
We were able to reproduce this issue with @jeffmaury. The problem was related to the `%USERPROFILE%/.ssh/config`. As a workaround we renamed the config file.

Can you please provide the output of the command `podman info --log-level debug`.

**Comment by glektarssza** [comment]
@l0rd Here you go:

```
time="2024-11-13T19:30:32-07:00" level=info msg="C:\\Users\\roblo\\scoop\\apps\\podman\\current\\podman.exe filtering at log level debug"
time="2024-11-13T19:30:32-07:00" level=debug msg="Called info.PersistentPreRunE(C:\\Users\\roblo\\scoop\\apps\\podman\\current\\podman.exe info --log-level debug)"
time="2024-11-13T19:30:32-07:00" level=debug msg="SSH Ident Key \"C:\\\\Users\\\\roblo\\\\.local\\\\share\\\\containers\\\\podman\\\\machine\\\\machine\" SHA256:DW4NTzZN7HgVBWBnF8rA6cmGd9LchILrC9GP0vnMU7Q ssh-ed25519"
time="2024-11-13T19:30:32-07:00" level=debug msg="DoRequest Method: GET URI: http://d/v5.3.0/libpod/_ping"
time="2024-11-13T19:30:32-07:00" level=debug msg="DoRequest Method: GET URI: http://d/v5.3.0/libpod/info"
host:
  arch: amd64
  buildahVersion: 1.37.5
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.12-2.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: '
  cpuUtilization:
    idlePercent: 99.47
    systemPercent: 0.35
    userPercent: 0.18
  cpus: 4
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: container
    version: "40"
  eventLogger: journald
  freeLocks: 2048
  hostname: GlekPC
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
  kernel: 5.15.167.4-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 16092475392
  memTotal: 16776024064
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.12.2-2.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.12.2
    package: netavark-1.12.2-1.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.12.2
  ociRuntime:
    name: crun
    package: crun-1.17-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.17
      commit: 000fa0d4eeed8938301f3bcf8206405315bc1017
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240906.g6b38f07-1.fc40.x86_64
    version: |
      pasta 0^20240906.g6b38f07-1.fc40.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 4294967296
  swapTotal: 4294967296
  uptime: 0h 6m 30.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 961994752
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 5.2.5
  Built: 1729814400
  BuiltTime: Thu Oct 24 18:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.7
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.5

time="2024-11-13T19:30:32-07:00" level=debug msg="Called info.PersistentPostRunE(C:\\Users\\roblo\\scoop\\apps\\podman\\current\\podman.exe info --log-level debug)"
time="2024-11-13T19:30:32-07:00" level=debug msg="Shutting down engines"
```

**Comment by l0rd** [comment]
Thank you. No we were not able to reproduce your error then. The problem was a different one related to the SSH  config.

We will try to reproduce it using scoop.

**Comment by johnnykang** [comment]
It looks like if Podman is installed via winget, win-sshproxy works perfectly fine.

I suspect this issue is specific to podman installation via scoop.

**Comment by glektarssza** [comment]
I'll try installing via `winget` after work today to confirm this as well. If I cannot reproduce the issue in that environment I think it's a scoop-specific issue, yeah.

**Comment by l0rd** [comment]
Thank you @johnnykang for checking. And you are right, that should be the problem. 

The version of [gvproxy has been updated](https://github.com/containers/podman/pull/24394) in 5.3 but [the scoop installer](https://github.com/ScoopInstaller/Main/blob/master/bucket/podman.json) extracts podman.exe, not gvproxy from the installer. 

@baude any idea who maintains the podman scoop bucket? From git history it looks @niheaven is the last one that have updated it.

**Comment by l0rd** [comment]
@glektarssza @johnnykang out of curiosity: why are you installing via scoop rather than winget?

**Comment by glektarssza** [comment]
I've been a long time user of `scoop` so mainly because it's an ecosystem I'm familiar with and, at the time I started using it, had the software I was using.

I'm not opposed to changing but it would be a bit of a pain to migrate everything over at this point depending on what is and is not available in the `winget` ecosystem.

**Comment by glektarssza** [comment]
Actually, I stand corrected. Their website links specifically to https://github.com/orgs/ScoopInstaller/people as the maintainers.

**Comment by jeffmaury** [comment]
I can reproduce after having installed Podman through scoop

**Comment by johnnykang** [comment]
due to the fact that this issue only occurs when Podman is installed via scoop, i would like to close this issue as i don't believe the team is responsible for fixing it. 

closing.

**Comment by glektarssza** [comment]
I opened ScoopInstalled/Main#6327. Hopefully they can figure this out on their end.

**Comment by niheaven** [comment]
Hi all, Scoop uses bundled version of `gvproxy.exe` and `win-sshproxy`, which version comes with `podman` v5.3.0?

And in my `podman` installation, `podman machine start` doesn't output any error, but yes, `podman machine stop` does, even after I replaced existing `gvproxy` and `win-sshproxy` in the `podman` folder with the latest ones (v0.8.0).

**Comment by johnnykang** [comment]
> And in my `podman` installation, `podman machine start` doesn't output any error

Mine output "The system cannot find the path specified." in the startup message. It is not an obvious message, but it warned me that the Docker API stopped working.

**Comment by niheaven** [comment]
Oh yes, same output, really not obvious :)

So which one should `podman` use? I'll try to install podman via winget and put all the files under scoop installed folder and see what will happened.

**Comment by johnnykang** [comment]
> Oh yes, same output, really not obvious :)
> 
> So which one should `podman` use? I'll try to install podman via winget and put all the files under scoop installed folder and see what will happened.

Just do scoop uninstall podman and winget install podman.


all configurations should still in effective and ready to use with all your existing containers. 

Try that at your own risk. It worked that that way on my machine ™️

**Comment by niheaven** [comment]
An error occurs when Podman is initiated from a Windows junction folder, as with Scoop's setup (where Podman starts from `podman\current`, a junction to `podman\5.3.0`). However, initiating Podman directly from the actual folder (`podman\5.3.0`) or from other locations like `C:\`, `D:\`, etc., works without issues.

Are there recent commits that change this behavior?

**Comment by l0rd** [comment]
@niheaven, we came to the same conclusion after doing some tests with @jeffmaury today. The workarounds we found:

- Run Podman from `<scoop-app-dir>\podman\5.3.0` as you mentioned above
- Create a `%APPDATA%\containers\containers.conf` with the `helper_binaries_dir` that points to `<scoop-app-dir>\podman\5.3.0`:
```toml
[engine]
helper_binaries_dir=["<scoop-app-dir>\podman\5.3.0"]
```
- Copy the `*.exe` files in `%PROGRAMFILES%/RedHat/Podman` (however, this requires admin privileges and may not be ideal)

When Podman starts from a junction it [fails to find the `win-sshproxy.exe`](https://github.com/containers/podman/blob/d3e7d4e2175d858d53d9142b0125fd38fc0dd83c/pkg/machine/machine_windows.go#L139) but it doesn't look that anything changed between 5.2 and 5.3 🤷

**Comment by l0rd** [comment]
> Is it necessary? I've noticed that the executables are identical between Winget and Scoop.

No, I don't think that's necessary. If you create the configuration file, you don't need to move the files there. 

> I'll try this and it should be done during Scoop installation.

Right, I think that's the best solution at the moment. We should figure out the root cause of the problem and eventually provide a fix (in Podman), but in the meantime, I think it's better to create the configuration file.

**Comment by niheaven** [comment]
> We should figure out the root cause of the problem and eventually provide a fix (in Podman)

Okay, let's wait for the fix, but for now, a hotfix in Scoop's manifest works quite well.

**Comment by niheaven** [comment]
Another question. If my scoop installation is at `D:\Scoop\apps\podman\5.3.0`, the config entry `helper_binaries_dir=["<scoop-app-dir>\podman\5.3.0"]` should be which format? 

- `helper_binaries_dir=["D:\Scoop\apps\podman\5.3.0"]`
- `helper_binaries_dir=["D:\\Scoop\\apps\\podman\\5.3.0"]`
- `helper_binaries_dir=["D:/Scoop/apps/podman/5.3.0"]`
- Or something else?

**Comment by cpfeiffer** [comment]
@johnnykang I think scoop is just the messenger here. There was [a change in go 1.23](https://github.com/containers/podman/blob/d3e7d4e2175d858d53d9142b0125fd38fc0dd83c/pkg/machine/machine_windows.go#L254) changing the behavior of `filePath.EvalSymlinks()` that may be related (podman 5.2  still used go 1.22).

**Comment by specter119** [comment]
> [@niheaven](https://github.com/niheaven), we came to the same conclusion after doing some tests with [@jeffmaury](https://github.com/jeffmaury) today. The workarounds we found:
> 
> * Run Podman from `<scoop-app-dir>\podman\5.3.0` as you mentioned above
> * Create a `%APPDATA%\containers\containers.conf` with the `helper_binaries_dir` that points to `<scoop-app-dir>\podman\5.3.0`:
> 
> [engine]
> helper_binaries_dir=["<scoop-app-dir>\podman\5.3.0"]
> * Copy the `*.exe` files in `%PROGRAMFILES%/RedHat/Podman` (however, this requires admin privileges and may not be ideal)
> 
> When Podman starts from a junction it [fails to find the `win-sshproxy.exe`](https://github.com/containers/podman/blob/d3e7d4e2175d858d53d9142b0125fd38fc0dd83c/pkg/machine/machine_windows.go#L139) but it doesn't look that anything changed between 5.2 and 5.3 🤷


I have a self-maintained scoop bucket, with podman 

<https://github.com/specter119/scoop-dsms/blob/main/bucket/podman.yml>

and tried 2 following solutions, neither works:

- `%APPDATA%\containers\containers.conf` 
- add gvproxy and win-sshproxy to PATH

I have fixed the issue in the official repo (like only extract podman.exe ) and tried the above approaches.

It seems that there are some remaining issues in the podman side, at least a proper config different with previous.

**Comment by l0rd** [comment]
A few more informations.

Indeed after [this change](https://github.com/golang/go/issues/63703) introduced in go v1.23, `filepath.EvalSymlinks(path)` returns an error if `path` is within a [junction ](https://learn.microsoft.com/en-us/sysinternals/downloads/junction) (as in the case of `podman.exe` installed via scoop). That's currently a Podman bug.

The good news is that setting the env `$env:GODEBUG="winsymlink=0"` when running Podman is sufficient to make it work as before. So that looks the simplest workaround @niheaven.

Othewise, the workarounds I [have mentioned above](https://github.com/containers/podman/issues/24557#issuecomment-2478100481), and in particular configuring `helper_binaries_dir`, work if the provider is Hyper-V. They don't when it's WSL. And WSL is the default provider and switching from WSL to HyperV can be problematic (a WSL machine cannot be converted to an HyperV machine). Anyway this is the `%APPDATA%\containers\containers.conf` I have tested with.

```ini
[engine]
helper_binaries_dir=["C:\\Users\\<username>\\scoop\\apps\\podman\\5.3.0\\"]

[machine]
provider="hyperv"
```

For WSL, the best workaround is to change `PATH` so that `C:\\Users\\<username>\\scoop\\apps\\podman\\5.3.0\\` comes first:

```pwsh
$env:PATH="C:\Users\<username>\scoop\apps\podman\5.3.0\;$env:PATH"
```

**Comment by l0rd** [comment]
I am re-opening this issue and assigning it to myself.

**Comment by niheaven** [comment]
Since I'm unsure whether other applications utilize `winsymlink` in Go, I implemented a temporary fix for Scoop manifest by adding the versioned directory to the PATH, which seems to be working well.

**Comment by Laess3r** [comment]
Hi guys!

Are there any updates to this topic? I had to instruct our team to not upgrade / downgrade to 5.2.5

Thanks and best regards.

**Comment by l0rd** [comment]
Hi @Laess3r and happy new year 👋 I should open a PR to fix this problem this week and hopefully the fix will land in v5.4.0.

Out of curiosity: what's the issue you are facing? I am asking because the original problem (after installing podman using scoop, machines fails to start) [should be fixed now](https://github.com/ScoopInstaller/Main/pull/6335) and I am wondering if there is another problem junction related problem we are not aware of.

**Comment by disacrol** [comment]
@l0rd I have the exact same issue as @Laess3r. Had to instruct the team to use a version lower than 5.3.
When I install podman from scoop and start a new podman machine, there is no error.
But if I stop podman machine and re-open the console, the error resurfaces when starting podman machine again.

Maybe there is some environment variable is use as a workaround, that is not persisted?

**Comment by glektarssza** [comment]
@disacrol Can you provide the contents of your `$env:PATH` after installing via `scoop` but before reopening PowerShell and then again after reopening PowerShell? That might shed some light on things.

**Comment by disacrol** [comment]
Indeed podman dir is not persisted in the path.
I noticed that scoop persists the path of some apps (e.g. python) but not others (e.g. notepadplusplus). Could this just be a matter of fixing some flag in the podman's scoop manifest or something? Or that https://github.com/ScoopInstaller/Main/pull/6335 was not enough to fix this?

Plain text to allow bold:

PS C:\Users\DOliveira> Write-Host $env:PATH
C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Users\DOliveira\scoop\shims;C:\Users\DOliveira\AppData\Local\Microsoft\WindowsApps;C:\Users\DOliveira\AppData\Local\Microsoft\WinGet\Packages\Microsoft.Sysinternals.TCPView_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\Users\DOliveira\AppData\Local\JetBrains\Toolbox\scripts
PS C:\Users\DOliveira> **scoop install podman**
Installing 'podman' (5.3.2) [64bit] from 'main' bucket
Loading podman-5.3.2-setup.exe from cache
Checking hash of podman-5.3.2-setup.exe ... ok.
Running installer script...Adding ~\scoop\apps\podman\5.3.2 to your path.
done.
Installer added '~\scoop\apps\podman\5.3.2' to path. Removing.
Linking ~\scoop\apps\podman\current => ~\scoop\apps\podman\5.3.2
Creating shim for 'podman'.
'podman' (5.3.2) was installed successfully!
PS C:\Users\DOliveira> Write-Host $env:PATH
**C:\Users\DOliveira\scoop\apps\podman\5.3.2**;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Users\DOliveira\scoop\shims;C:\Users\DOliveira\AppData\Local\Microsoft\WindowsApps;C:\Users\DOliveira\AppData\Local\Microsoft\WinGet\Packages\Microsoft.Sysinternals.TCPView_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\Users\DOliveira\AppData\Local\JetBrains\Toolbox\scripts

(reopen PS)

PS C:\Users\DOliveira> Write-Host $env:PATH
C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Users\DOliveira\scoop\shims;C:\Users\DOliveira\AppData\Local\Microsoft\WindowsApps;C:\Users\DOliveira\AppData\Local\Microsoft\WinGet\Packages\Microsoft.Sysinternals.TCPView_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\Users\DOliveira\AppData\Local\JetBrains\Toolbox\scripts

**Comment by glektarssza** [comment]
Yeah, it's strange that it's not being persisted to your path... This is definitely something to do with `scoop` as I'm seeing the same behaviour on my system too. This is not a `podman` issue, I think. Best to report it to the `scoop` team.

**Comment by niheaven** [comment]
Scoop adds something to PATH in two steps: 1. it adds it to the current session; 2. it adds it via Environment Variables. Please check if it works after a reboot (to ensure Environment Variables are applied).

**Comment by disacrol** [comment]
Yes it does - and now that I have setup freshly (with newer podman 5.4.0 btw), it also works after re-opening PowerShell. Maybe there was something odd in my system. Thanks @l0rd and @niheaven for the info!

**Comment by l0rd** [comment]
With 5.4.0 the [fix](https://github.com/ScoopInstaller/Main/pull/6335) provided by @niheaven (the `PATH` update) should not be necessary anymore. I have created a [PR ](https://github.com/ScoopInstaller/Main/pull/6528) to revert it.
